### ISSUE TITLE
Auto Update media Libray when changes is detected

### ISSUE THREAD

**Issue by felipesalomao** [boxeehacks/boxeehack#75]
Hi, would be great auto update media libray when monitored folders have some change and not need to wait 1 hour to update libray or scan manually and wait about 2 minutes to find new content.. For example, when we transfer tv program like dexter to monitored folder, (my boxee is aways powered on), i have to scan and wait few minutes to be able to watch tv program epsode..

**Comment by kelmk** [comment]
+1

**Comment by Silent337** [comment]
+1

**Comment by boxeehacks** [comment]
I have some ideas for this!

**Comment by jerbob92** [comment]
Can we use inotify on the box? Not sure if it would work for SMB mounts though.

**Comment by felipesalomao** [comment]
Somebody is developing it ?

**Comment by boxeehacks** [comment]
SMB is not accessed in a standard way though. Some stuff is access through mounting, but some stuff is done directly, and there is stuff cached on the box itself. So just using inotify over the mounts won't help very much (they may not be mounted at all).

However I have added a new remote control mechanism to Boxee+ that you can use to externally trigger a rescan!

I'll provide some documentation for that in the future. But the new version will ship with this, and has a sample implementation as a growl plugin that can send notifications to the BB. Check the Wiki in a few days...

**Comment by brdhse1** [comment]
@boxeehacks 
Is this the new functionality that you are referring to?  http://developer.boxee.tv/Remote_Control_Interface
If this is it, is there a username/password required?  Is it listening on port 9090?  I can't seem to make a simple request like this work:  curl "http://192.168.1.12:9090/xbmcCmds/xbmcHttp?command=Mute"

Edit:
Ok, I found it listening on port 8800, but I can't get any of the ExecBuiltIn() commands to work - therefore no rescan.  What am I doing wrong?

Edit 2:
I found boxeehack_server.py listening on port 2100 and I see that it calls xbmc.executebuiltin with whatever was sent to it.  I can get basics like Mute to work through here, but not UpdateLIbrary (http://192.168.1.12:2100/UpdateLibrary(video))
